core/src/main/java/org/ehcache/events/CacheEventNotificationServiceImpl.java

```diff
@@ -122,112 +174,162 @@ public void setStoreListenerSource(Cache<K, V> source) {
   }
 
   @Override
-  public void onEvent(final CacheEvent<K, V> event) {
+  public void onEvent(CacheEvent<K, V> event) {
```

Comment:
Objective – Process event so its in-line for dispatch depending on ordered/unordered dispatch.
What it internally does:
•	Add event to threadLocal.eventList and to the correct event queue if events are ordered.

Why it should work:
The premise here is that each thread should fire the events it creates. This is done by maintaining a list of events created by the thread in a list in its threadLocal. At the time of dispatch, if ordered, thread checks its list of events, visits each concerned queue to dispatch the events. Else, in case of unordered, a dispatch task needs to be created directly.
Thus in this method, in preparation for the dispatch, the event is added to the threadLocal.eventList and the eventQueues in case of ordered dispatch.